### ISSUE TITLE
#357 breaks projects with `exactOptionalPropertyTypes` enabled

### ISSUE THREAD

**Issue by dhess** [anymaniax/orval#442]
The addition of query cancelation in #357 is welcome, but the current implementation breaks TypeScript projects with `exactOptionalPropertyTypes` enabled:

```
error TS2379: Argument of type '{ url: string; method: string; signal: AbortSignal | undefined; params: GetSessionListParams | undefined; }' is not assignable to parameter of type 'AxiosRequestConfig<any>' with 'exactOptionalPropertyTypes: true'. Consider adding 'undefined' to the types of the target's properties.
```

**Comment by anymaniax** [comment]
Hello @dhess, I will change the syntax like this to solve the problem

```ts
export const listPets = (
  params?: ListPetsParams,
  version = 1,
  signal?: AbortSignal,
) => {
  return customInstance<Pets>({
    url: `/v${version}/pets`,
    method: 'get',
    ...(params ? { params } : {}),
    ...(signal ? { signal } : {}),
  });
};
```

**Comment by anymaniax** [comment]
I did a fix for that will be in the next release

**Comment by anymaniax** [comment]
Released with 6.9.0

**Comment by dhess** [comment]
Thanks. For the record, we still get this error in 6.9.1 if we have `exactOptionalPropertyTypes' enabled and use the default Orval `query` settings. However, if we implement the override feature you described [here](https://github.com/anymaniax/orval/issues/395#issuecomment-1186189545) as so, then everything works again, so this will suffice:

```
  query: {
    signal: false,
  },
```
